src/d_main.c

```diff
@@ -2148,9 +2162,16 @@ void D_DoomMain(void)
 
   if ((p = M_CheckParm ("-episode")) && p < myargc-1)
     {
-      startepisode = myargv[p+1][0]-'0';
-      startmap = 1;
-      autostart = true;
+      if (M_ParmStrToInt(myargv[p+1], &startepisode) &&
+          startepisode >= 1 && startepisode <= 4)
+      {
+        startmap = 1;
+        autostart = true;
+      }
+     else
+      {
+        I_Error("Wrong -episode parameter '%s', should be 1-4", myargv[p+1]);
```

Comment:
Can't we have any episode from 0 to 9 now?

Reply:
Right, in Woof we even support episodes 0-99.
@fabiangreffrath BTW `-warp 9999` doesn't currently work, only `-warp 99 99`. Not sure how to validate `-warp` 😄

Reply:
I think it's okay as it is. No algorithm can know if `-warp 999` means to warp to E99M9 or E9M99. 😉